### ISSUE TITLE
Allow pod-utils / clonerefs to use default remote branch

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/test-infra#20667]
My read of current pod-utils code is that it assumes `base_ref` to always be present / non-empty (ref: https://github.com/kubernetes/test-infra/pull/20665#issuecomment-769961333) 

It looks like there are 435 jobs that explicitly reference `master`, when what they likely mean is "just use whatever the default remote branch is"
```
$ ag --no-filename "base_ref:.*master" config/jobs | grep base_ref | wc -l
     435
```

Ideally base_ref would be auto-populated with the remote default branch if omitted.  Maybe this already happens?  This would make repo default branch renames less toilsome.

Opening this to track verifying or fixing, so not yet a bug or feature.

/area prow/pod-utilities

/assign @alvaroaleman @cjwagner 
in case you can answer more quickly than I can, feel free to /unassign if you can't handle
/assign @spiffxp 
to verify what happens today

**Comment by alvaroaleman** [comment]
> My read of current pod-utils code is that it assumes base_ref to always be present / non-empty (ref: #20665 (comment))

Yeah, that is correct (for periodics, for presubmits/possubmits we get the info from github)

> Ideally base_ref would be auto-populated with the remote default branch if omitted. Maybe this already happens? This would make repo default branch renames less toilsome.

I am pretty sure this does not happen. Adding this would be pretty easy from a complexity POV, as the `default_branch` is part of the repo. It will drive up token usage a bit though (and I think it won't cache _that_ well, as the repo contains a lot of info that will regularly change, like `pushed_at: timestamp`, size, the number of stars, forks etc)

**Comment by spiffxp** [comment]
If we're concerned about token usage, we could try doing this directly in clone by using the contents of  `.git/refs/remotes/origin/HEAD` if `base_ref` is empty.  I dunno if that makes the API feel too weird though (technically it does `omitempty` as-is)

**Comment by spiffxp** [comment]
/area jobs
since this would be a job config change too

**Comment by spiffxp** [comment]
/wg naming
since this is in support of https://github.com/kubernetes/org/issues/2222

**Comment by spiffxp** [comment]
Relevant discussion thread in #prow https://kubernetes.slack.com/archives/CDECRSC5U/p1615236463055600

**Comment by stevekuznetsov** [comment]
Since the `HEAD` symbolic ref is not a default `git` thing and instead a real symbolic ref that e.g. GitHub, Gerrit, GitLab provide, it should be possible to just use `extra_refs[*].base_ref: HEAD`.

**Comment by spiffxp** [comment]
FWIW, I feel like remotes having a `HEAD` is a default `git` thing, but maybe I'm wrong.  I'll give the `base_ref: HEAD` thing a shot this week

refs:
- https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes
- http://git.661346.n2.nabble.com/What-s-the-definition-of-a-valid-Git-symbolic-reference-tt6025154.html#a6026446

**Comment by BenTheElder** [comment]
not quite workable, because we try to check it out to a branch with that name, and it's not a valid branch name https://github.com/kubernetes/test-infra/pull/21452#issuecomment-803075138

**Comment by BenTheElder** [comment]
I think we could add something simple to replace HEAD with a valid branch name and should still consider this.

**Comment by spiffxp** [comment]
```
$ ag --no-filename "base_ref:.*master" config/jobs | grep base_ref | wc -l
    1141
```
The number of jobs affected has grown considerably from the 435 that existed at the beginning of this year

**Comment by BenTheElder** [comment]
@cjwagner @stevekuznetsov @chaodaiG WDYT https://github.com/kubernetes/test-infra/issues/20667#issuecomment-863556005 ?

I think we could do this pretty easily by just detecting `HEAD` => pick a different value for the branch name but otherwise use HEAD in most places untouched. Last time I tried this the branch name seemed to be the only stumbling block. I'm not sure what we should call it.

**Comment by BenTheElder** [comment]
This would now be at https://sigs.k8s.io/prow

I think so far we've just wound up updating the configs as necessary. I do think this would be nice to have, but there clearly haven't been the resources and this is no longer the place to track prow features.
